embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/session/orchestrator/SessionOrchestratorImpl.kt

```diff
@@ -265,15 +277,11 @@
         endProcessState: ProcessState
     ): Envelope<SessionPayload>? {
         Systrace.traceSynchronous("on-session-cache") {
-            return if (initial.sessionId == sessionIdTracker.getActiveSessionId()) {
```

Comment:
Are we still at risk of caching data for a session that doesn't match the sessionId in the `initial` payload?

Reply:
I've moved this check to elsewhere within `SessionOrchestratorImpl`.